xarray/core/dataset.py

```diff
@@ -9066,15 +9065,7 @@ def polyfit(
         variables = {}
         skipna_da = skipna
 
-        x: Any = self.coords[dim].variable
-        x = _floatize_x((x,), (x,))[0][0]
-
-        try:
-            x = x.values.astype(np.float64)
-        except TypeError as e:
-            raise TypeError(
-                f"Dim {dim!r} must be castable to float64, got {type(x).__name__}."
-            ) from e
+        x: Any = _ensure_numeric(self.coords[dim]).astype(np.float64)
```

Comment:
Is the Any required?
What is the mypy error when it is removed?

Reply:
agree, can we use `#type: ignore` instead?

Reply:
Honestly, the `Any` was there because I simply left it there...

If I remove it , the `mypy` error is triggered below at line 9072, on `np.vander` and reads:
 ```
 xarray/core/dataset.py: note: In member "polyfit" of class "Dataset":
xarray/core/dataset.py:9072: error: Argument 1 to "vander" has incompatible type "Dataset | DataArray"; expected "_SupportsArray[dtype[numpy.bool] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int]"  [arg-type]
```

I tried adding `.values` and typing as a `ndarray`, but that didn't really help as `_ensure_numeric` has  `Dataset` in its output types (even though it is impossible here). So I am adding `type: ignore`.

Reply:
In fact, `np.asarray(x)` seems to fix the typing issues.

Reply:
Not sure if it is important to keep the indexes (and coords) here.

But also, I don't know what `np.vander` does with e.g. a `xr.DataArray`...

Maybe a type ignore is easier by now.